openssl/aes.go

```diff
@@ -374,28 +303,30 @@ func (c *aesCipher) NewGCMTLS() (cipher.AEAD, error) {
 }
 
 func (c *aesCipher) newGCM(tls bool) (cipher.AEAD, error) {
-	keyLen := len(c.key) * 8
-
-	if keyLen != 128 && keyLen != 256 {
-		if ExecutingTest() {
-			// Fall back to standard library for GCM with non-standard key size.
-			return cipher.NewGCMWithNonceSize(&noGCM{c}, gcmStandardNonceSize)
-		}
-		// Return error for GCM with non-standard key size.
-		return nil, fail("GCM invoked with non-standard key size")
-	}
-
-	g := &aesGCM{key: c.key, tls: tls}
-	if g.NonceSize() != gcmStandardNonceSize {
-		panic("boringcrypto: internal confusion about nonce size")
+	var cipher C.GO_EVP_CIPHER_PTR
+	switch len(c.key) * 8 {
+	case 128:
+		cipher = C.go_openssl_EVP_aes_128_gcm()
+	case 192:
```

Comment:
192-bit keys are not supported by RedHat. This restirction should me moved to the integration with Go (i.e, in the crypto backend).